receiver/awscontainerinsightreceiver/internal/k8sapiserver/k8sapiserver.go

```diff
@@ -16,54 +16,270 @@ package k8sapiserver
+	clusterName := k.clusterNameProvider.GetClusterName()
+	if clusterName == "" {
+		k.logger.Warn("Failed to detect cluster name. Drop all metrics")
+		return result
+	}
+
+	if k.leading {
+		k.logger.Info("collect data from K8s API Server...")
+		timestampNs := strconv.FormatInt(time.Now().UnixNano(), 10)
+		client := k.k8sClient
+
+		fields := map[string]interface{}{
+			"cluster_failed_node_count": client.Node.ClusterFailedNodeCount(),
+			"cluster_node_count":        client.Node.ClusterNodeCount(),
+		}
+		attributes := map[string]string{
+			ci.ClusterNameKey: clusterName,
+			ci.MetricType:     ci.TypeCluster,
+			ci.Timestamp:      timestampNs,
+			ci.Version:        "0",
+		}
+		if k.nodeName != "" {
+			attributes["NodeName"] = k.nodeName
+		}
+		md := ci.ConvertToOTLPMetrics(fields, attributes, k.logger)
+		result = append(result, md)
+
+		for service, podNum := range client.Ep.ServiceToPodNum() {
+			fields := map[string]interface{}{
+				"service_number_of_running_pods": podNum,
+			}
+			attributes := map[string]string{
+				ci.ClusterNameKey: clusterName,
+				ci.MetricType:     ci.TypeClusterService,
+				ci.Timestamp:      timestampNs,
+				ci.TypeService:    service.ServiceName,
+				ci.K8sNamespace:   service.Namespace,
+				ci.Version:        "0",
+			}
+			if k.nodeName != "" {
+				attributes["NodeName"] = k.nodeName
+			}
+			md := ci.ConvertToOTLPMetrics(fields, attributes, k.logger)
+			result = append(result, md)
+		}
+
+		for namespace, podNum := range client.Pod.NamespaceToRunningPodNum() {
+			fields := map[string]interface{}{
+				"namespace_number_of_running_pods": podNum,
+			}
+			attributes := map[string]string{
+				ci.ClusterNameKey: clusterName,
+				ci.MetricType:     ci.TypeClusterNamespace,
+				ci.Timestamp:      timestampNs,
+				ci.K8sNamespace:   namespace,
+				ci.Version:        "0",
+			}
+			if k.nodeName != "" {
+				attributes["NodeName"] = k.nodeName
+			}
+			md := ci.ConvertToOTLPMetrics(fields, attributes, k.logger)
+			result = append(result, md)
+		}
+	}
 	return result
 }
 
-func (k *K8sAPIServer) start() error {
-	//TODO: add implementation
+func (k *K8sAPIServer) init() error {
+	var ctx context.Context
+	ctx, k.cancel = context.WithCancel(context.Background())
+
+	k.nodeName = os.Getenv("HOST_NAME")
+	if k.nodeName == "" {
+		return errors.New("missing environment variable HOST_NAME. Please check your deployment YAML config")
+	}
+
+	lockNamespace := os.Getenv("K8S_NAMESPACE")
+	if lockNamespace == "" {
+		return errors.New("missing environment variable K8S_NAMESPACE. Please check your deployment YAML config")
+	}
+
+	configMapInterface := k.k8sClient.ClientSet.CoreV1().ConfigMaps(lockNamespace)
+	if configMap, err := configMapInterface.Get(ctx, lockName, metav1.GetOptions{}); configMap == nil || err != nil {
+		k.logger.Info(fmt.Sprintf("Cannot get the leader config map: %v, try to create the config map...", err))
+		configMap, err = configMapInterface.Create(ctx,
```

Comment:
Creating a k8s resource seems to invasive for the collector isn't it? Can it be cleaned up, etc?

Reply:
The `configmap` is created in the given namespace. When the namespace is deleted, everything in the namespace will be deleted. 

So they will be cleaned up when users delete the collector using the yaml file that we provided.

To deploy container insight
```
kubectl apply -f yaml.file
```

To delete 
```
kubectl delete -f yaml.file
```